### PR TITLE
IDEX-2848: added meaningful docker container name for machine

### PR DESCRIPTION
**Pull request by akorneta**

### CODE REVIEW

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 1]
```diff
@@ -362,4 +368,18 @@ private Instance createInstance(String imageId,
...
+
+    String generateContainerName(String workspaceId, String displayName, String machineId) throws MachineException {
+        String workspaceName;
+        String userName = EnvironmentContext.getCurrent().getUser().getName();
+        try {
+            workspaceName = runtimeWorkspaceRegistry.get(workspaceId).getName();
+        } catch (NotFoundException e) {
+            throw new MachineException("Impossible to find running workspace for machine " + machineId, e);
+        }
+        workspaceName = workspaceName == null ? "" : workspaceName;
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: Actually <code>workspaceName</code> is always not null.
<code>RuntimeWorkspaceRegistry</code> either returns <code>RuntimeWorkspace</code> instance or throws <code>NotFoundException</code>.

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 2]
```diff
@@ -362,4 +368,18 @@ private Instance createInstance(String imageId,
             throw new MachineException(e);
         }
     }
+
+    String generateContainerName(String workspaceId, String displayName, String machineId) throws MachineException {
+        String workspaceName;
+        String userName = EnvironmentContext.getCurrent().getUser().getName();
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: <code>userName</code> is defined 6 lines above the place it is used

File: plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java [hunk 3]
```diff
@@ -6,7 +6,7 @@
  * http://www.eclipse.org/legal/epl-v10.html
  *
  * Contributors:
- *   Codenvy, S.A. - initial API and implementation
```
Comment (evoevodin) [inline plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java]: :scream:

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java (hunk 1 above)
Comment (akorneta) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: :+1:

File: plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java (hunk 3 above)
Comment (akorneta) [inline plugin-docker/che-plugin-docker-machine/src/test/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProviderTest.java]: :scream:

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 4]
```diff
@@ -363,4 +369,19 @@ private Instance createInstance(String imageId,
...
+
+    String generateContainerName(String workspaceId, String displayName) throws MachineException {
+        String workspaceName;
+        try {
+            workspaceName = runtimeWorkspaceRegistry.get(workspaceId).getName();
+        } catch (NotFoundException e) {
+            throw new MachineException("Impossible to find running workspace for machine ", e);
+        }
+        String userName = EnvironmentContext.getCurrent().getUser().getName();
+        /* container name is userName + workspaceName + displayName separated with '_' */
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: useless comment, code says the same. BTW `//` preceded one-line comment

File: plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java [hunk 5]
```diff
@@ -363,4 +369,19 @@ private Instance createInstance(String imageId,
...
+        final String containerName = userName + '_' + workspaceName + '_' + displayName + '_';
+
+        /* replacing not allowed characters + generating random name suffix */
```
Comment (garagatyi) [inline plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java]: I think it is removal not replacement

**Comment by garagatyi** [comment]
ok

**Comment by evoevodin** [comment]
:+1:

**Pull request merged**